ios/sdk/WeexSDK/Sources/Utility/WXBoxShadow.m

```diff
@@ -0,0 +1,130 @@
+/**
+ * Created by Weex.
+ * Copyright (c) 2016, Alibaba, Inc. All rights reserved.
+ *
+ * This source code is licensed under the Apache Licence 2.0.
+ * For the full copyright and license information,please view the LICENSE file in the root directory of this source tree.
+ */
+
+#import "WXBoxShadow.h"
+#import "WXConvert.h"
+
+@implementation WXBoxShadow
+
++(void)configLayer:(UIView *)view boxShadow:(WXBoxShadow *)boxShadow
+{
+    if(!boxShadow){
+        return;
+    }
+    if (boxShadow.isInset) {
+        if (boxShadow.innerLayer) {
+            [boxShadow.innerLayer removeFromSuperlayer];
+            boxShadow.innerLayer.frame = view.bounds;
+            [view.layer addSublayer:boxShadow.innerLayer];
+        }
+    } else {
+        UIBezierPath *shadowPath = [UIBezierPath bezierPathWithRect:view.bounds];
+        view.layer.masksToBounds = NO;
+        view.layer.shadowColor = boxShadow.shadowColor;
+        view.layer.shadowOffset = boxShadow.shadowOffset;
+        view.layer.shadowRadius = boxShadow.shadowRadius;
+        view.layer.shadowOpacity = boxShadow.shadowOpacity;
+        view.layer.shadowPath = shadowPath.CGPath;
+    }
+}
+
++ (NSArray *)getBoxShadowElementsByBlank:(NSString *)string
+{
+    string = [string stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
+    NSError *error = nil;
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:@"\\s{2,}" options:NSRegularExpressionCaseInsensitive error:&error];
+    
+    NSArray *arr = [regex matchesInString:string options:NSMatchingReportCompletion range:NSMakeRange(0, [string length])];
+    
+    arr = [[arr reverseObjectEnumerator] allObjects];
+    for (NSTextCheckingResult *str in arr) {
+        string = [string stringByReplacingCharactersInRange:[str range] withString:@" "];
+    }
+    NSArray *array = [string componentsSeparatedByString:@" "];
+    return array;
+}
+
++(WXBoxShadow *)getBoxShadowFromString:(NSString *)string
+{
+    if ([string length] == 0) {
+        return nil;
+    }
+    WXBoxShadow *boxShadow = [WXBoxShadow new];
+    boxShadow.shadowOpacity = 1.0f;
+    
+    //parse color
+    if ([string rangeOfString:@"rgb"].location != NSNotFound) {
+        NSRange begin = [string rangeOfString:@"rgb"];
+        NSRange end = [string rangeOfString:@")"];
+        if (begin.location < end.location) {
```

Comment:
`begin.location` is ok but end.location  may be NSNotFound which is a max number.